### PR TITLE
Add test validating collections drop references

### PR DESCRIPTION
**Pull request by stephentoub**
https://github.com/dotnet/runtime/pull/50182#issuecomment-806204104

### CODE REVIEW

**Comment by VSadov** [comment]
Is it possible to do something similar for dictionaries? 
In particular - removal of a key should drop both the key and the value.

**Comment by stephentoub** [comment]
> Is it possible to do something similar for dictionaries?

Done

**Review by VSadov — APPROVED** [review]
LGTM!

File: src/libraries/Common/tests/System/Collections/ICollection.Generic.Tests.cs [hunk 1]
```diff
@@ -334,6 +336,46 @@ public void ICollection_Generic_Clear_Repeatedly(int count)
             }
         }
 
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsPreciseGcSupported))]
```
Comment (danmoseley) [inline src/libraries/Common/tests/System/Collections/ICollection.Generic.Tests.cs:339]: are we covered for TrimExcess() (where available)?

Comment (danmoseley) [reply src/libraries/Common/tests/System/Collections/ICollection.Generic.Tests.cs:339]: On all of Stack, Queue, List, HashSet, Dictionary

**Review by danmoseley — APPROVED** [review]

File: src/libraries/Common/tests/System/Collections/ICollection.Generic.Tests.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/Common/tests/System/Collections/ICollection.Generic.Tests.cs:339]: I don't think TrimExcess is relevant for this.  It's shrinking the size of the array after items have already been removed; it's the removal that's impacted, not the subsequent shrinking.

Comment (danmoseley) [reply src/libraries/Common/tests/System/Collections/ICollection.Generic.Tests.cs:339]: Yes I don't know what o was thinking 😊

**Pull request merged**